build/ci_utils/src/cache/goodie/graalvm.rs

```diff
@@ -88,36 +79,37 @@ impl GraalVM {
     pub fn url(&self) -> BoxFuture<'static, Result<Url>> {
         let platform_string = self.platform_string();
         let graal_version = self.graal_version.clone();
+        let graal_version_tag = format!("{}.{}.{}", graal_version.major, graal_version.minor, graal_version.patch);
         let client = self.client.clone();
         async move {
             let repo = CE_BUILDS_REPOSITORY.handle(&client);
-            let release = repo.find_release_by_text(&graal_version.to_string()).await?;
+            let release = repo.find_release_by_text(&graal_version_tag).await?;
             crate::github::find_asset_url_by_text(&release, &platform_string).cloned()
         }
         .boxed()
     }
 
     pub fn platform_string(&self) -> String {
-        let Self { graal_version: _graal_version, java_version, arch, os, client: _client } = &self;
+        let Self { graal_version: _graal_version, arch, os, client: _client } = &self;
         let os_name = match *os {
             OS::Linux => "linux",
             OS::Windows => "windows",
             OS::MacOS => "darwin",
             other_os => unimplemented!("System `{}` is not supported!", other_os),
         };
         let arch_name = match *arch {
-            Arch::X86_64 => "amd64",
-            // No Graal packages for Apple Silicon.
-            Arch::AArch64 if TARGET_OS == OS::MacOS => "amd64",
+            Arch::X86_64 => "x64",
             Arch::AArch64 => "aarch64",
```

Comment:
Are you trying to fix #6862 by downloading `aarch64` on Mac M1? Fixing #6862 would be desirable, but I doubt one can fix that without having access to Mac M1 hardware.

Also, I'd like to avoid letting that issue block upgrade to GraalVM 23.0.0 JDK 17.

Reply:
No, sorry for the confusion. I rolled back this change in 83ca38475c51168ea7d1257d532e5a6208208d4e - the only change here is the renaming from `amd64` to `x64`. I won't touch M1-specific parts.